ring/model.go

```diff
@@ -602,16 +602,42 @@ func MergeTokens(instances [][]uint32) []uint32 {
 		numTokens += len(tokens)
 	}
 
-	tree := loser.New(instances, math.MaxUint32)
+	lists := make([]*sliceSequence, len(instances))
+	for i := range instances {
+		lists[i] = &sliceSequence{s: instances[i]}
+	}
+	tree := loser.New[uint32](lists, math.MaxUint32)
 	out := make([]uint32, 0, numTokens)
 
 	for tree.Next() {
-		out = append(out, tree.Winner())
+		out = append(out, tree.At())
 	}
 
 	return out
 }
 
+// Wrapper over a slice that implements the loser.Sequence API
```

Comment:
Is it worth including a version of this in the `loser` package? I imagine this won't be the only place that needs to work with slices.

Having said that, I'm not sure whether we'd want a generic slice sequence type, or specialised types - would be interesting to benchmark and see if the use of generics makes any substantial performance difference.